subsys/bluetooth/host/gatt.c

```diff
@@ -5158,7 +5170,9 @@ void bt_gatt_connected(struct bt_conn *conn)
 
 #if defined(CONFIG_BT_GATT_CLIENT)
 	add_subscriptions(conn);
-
+#if defined(CONFIG_BT_GATT_AUTO_UPDATE_MTU)
+	bt_gatt_exchange_mtu(conn, &gatt_exchange_params);
```

Comment:
Since `bt_gatt_exchange_mtu` can fail we should either handle that or at the very least log it.